### PR TITLE
Home Page 

### PR DESCRIPTION
**Pull request by OnsNader**
Made home hbs and it's corresponding stylesheet. Made it's route and the renderpage and it's test.
Relate #8

### CODE REVIEW

File: public/css/home.css [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    margin: 0 auto;
+    display:flex;
+    justify-content: space-between;
+    margin-bottom: 10px;
+}
+main section .container3 .pic {
+    width: 50%;
+    height: 300px;
+    border: 1px solid black;
+    /* margin: 10px 20px; */
```
Comment (knanahassouna1) [inline public/css/home.css]: remove commented code .
```  
  /* margin: 10px 20px; */
```

File: src/controllers/home.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+exports.get = (req, res) => {
+    res.render("home", {
+        title: "chat it out", headerFound: true, footerFound: true, style: ["home", "footer", "header"],
+        javascript: ['home']
```
Comment (mohannadhanafi) [inline src/controllers/home.js]:
```
javascript: ['home', 'hamburger']
```
Add hamburger to javascript

File: public/css/home.css [hunk 3]
```diff
@@ -0,0 +1,307 @@
+*{
+    margin: 0;
+    padding: 0;
+}
```
Comment (mohannadhanafi) [inline public/css/home.css]:
```
*{
    margin: 0;
    padding: 0;
}
```
Remove this and link the public.css file

File: src/views/home.hbs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                        
+                    </label>
+                    
+                    <button>Send</button>
+
+                </div>
+
+            </div>
+        </section>
+
```
Comment (knanahassouna1) [inline src/views/home.hbs]: remove white lines .

File: src/controllers/index.js [hunk 5]
```diff
@@ -1,6 +1,11 @@
 const Router = require('express').Router();
 const chat = require('./chat');
-
+const home = require('./home');
+const signUp = require ('./signUp');
```
Comment (MohammedYehia) [inline src/controllers/index.js]: signin/up should not be exited in this pull request and also I don't see any file called signup.js or signin!!

File: src/views/home.hbs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            </div>
+        </div>
+    </div>
+    <!-- Thereby conculdes the showcase section -->
+
+    <!-- Our services & contact us sections -->
+    <main>
+        <section>
+            <h1>Our Services</h1>
+            <div class="container2">
```
Comment (MohammedYehia) [inline src/views/home.hbs]: try to avoid using class name like container 1,2,3...

File: src/views/home.hbs [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                <div class="item">
```
Comment (MohammedYehia) [inline src/views/home.hbs]: from this line 22 -47 you are using the same pattern where div>h1>p
with express handlebars you can use each loop and pass all titles and the paragraphs with object to the hbs file
```
{{#each obj}}
<div class="item"/>
    <h1>{{this.title}}</h1>
    <p>{{this.content}}</p>
</div>
{{/each}}
```

File: test/test_BE.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    supertest(app)
+      .get('/')
+      .expect(200)
+      .end((err, res) => {
+        if (err) {
+          t.error(err);
+        }
+        t.ok(res.text.includes('body'));
+        t.equal(res.text.substr(0, 15), '<!DOCTYPE html>', 'the respone  should be html file');
+        t.equal(res.header['content-type'], 'text/html; charset=utf-8', 'the content-type should equal html');
```
Comment (MohammedYehia) [inline test/test_BE.js]: you can use `.expect('Content-Type', /html/)`

File: test/test_BE.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        t.end();
+      });
+  });
+  //test the sign-in route
```
Comment (MohammedYehia) [inline test/test_BE.js]: Is the sign up/ in part of the home issue?
remove these tests  and upload it with its own issue

File: src/views/home.hbs (hunk 7 above)
Comment (OnsNader) [reply src/views/home.hbs]: WIll do in later projects

File: src/views/home.hbs (hunk 6 above)
Comment (OnsNader) [reply src/views/home.hbs]: Ookay

File: npm-debug.log [hunk 10]
```diff
@@ -0,0 +1,45 @@
+0 info it worked if it ends with ok
```
Comment (MohammedYehia) [inline npm-debug.log]: Remove this file

File: test/test_BE.js [hunk 11]
```diff
@@ -14,35 +15,7 @@ tape('test for home route ', (t) => {
         t.equal(res.text.substr(0, 15), '<!DOCTYPE html>', 'the respone  should be html file');
         t.equal(res.header['content-type'], 'text/html; charset=utf-8', 'the content-type should equal html');
```
Comment (MohammedYehia) [inline test/test_BE.js]: You don't need line 16 `t.equal(res.header['content-type'], 'text/html; charset=utf-8', 'the content-type should equal html');`  after using `.expect('Content-Type', /html/)`
so remove line 16

File: src/views/home.hbs (hunk 7 above)
Comment (MohammedYehia) [reply src/views/home.hbs]: @OnsNader **Then we will merge in later projects**

**Pull request merged**